File: 2022/283. Move Zeroes/Solution2.java
class Solution {
    public void moveZeroes(int[] nums) {
        if (nums == null || nums.length <= 1)
            return;
        int nonzero = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0)
                nums[nonzero++] = nums[i]; 
        }
        while (nonzero < nums.length) {
            nums[nonzero++] = 0;
        }
    }
}

File: 2025/45. Jump Game II/solution2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [float('inf')] * (n)
        dp[-1] = 0
        for i in range(n - 2, -1, -1):
            max_jump = min(nums[i], n - i - 1)
            for j in range(1, max_jump + 1):
                dp[i] = min(dp[i + j] + 1, dp[i])
        return dp[0]